src/backend/linux_raw/termios/syscalls.rs

```diff
@@ -238,14 +239,19 @@ pub(crate) fn ttyname(fd: BorrowedFd<'_>, buf: &mut [u8]) -> io::Result<usize> {
     if r == buf.len() {
         return Err(io::Errno::RANGE);
     }
-    buf[r] = b'\0';
+    buf[r].write(b'\0');
 
     // Check that the path we read refers to the same file as `fd`.
-    let path = CStr::from_bytes_with_nul(&buf[..=r]).unwrap();
+    {
+        // SAFETY:
+        // - We just wrote the NUL byte above
+        // - readlinkat() returns the number of bytes placed in the buffer
```

Comment:
As a minor style nit, I prefer to see function names code-quoted and not suffixed with `()` unless they actually have zero arguments.  The old `()` convention reminds me too much of old-school K&R C where you could just `()` even if the function has args :-}.

Reply:
Lol, fixed.